src/adapters/morpho-aave-v2/common/morphoBasePoolAdapter.ts

```diff
@@ -0,0 +1,633 @@
+          ])
+
+          return {
+            ...protocolTokenMetadata,
+            type: TokenType.Protocol,
+            tokens: Object.values(underlyingEndPositions).map(
+              ({
+                address,
+                name,
+                symbol,
+                decimals,
+                balanceRaw: endPositionValueRaw,
+              }) => {
+                const startPositionValueRaw =
+                  startPositionValues[protocolTokenMetadata.address]
+                    ?.underlyingTokenPositions[address]?.balanceRaw ?? 0n
+
+                const calculationData = {
+                  withdrawalsRaw: withdrawals[address] ?? 0n,
+                  depositsRaw: deposits[address] ?? 0n,
+                  endPositionValueRaw: endPositionValueRaw ?? 0n,
+                  startPositionValueRaw,
+                }
+
+                let profitRaw =
+                  calculationData.endPositionValueRaw +
+                  calculationData.withdrawalsRaw -
+                  calculationData.depositsRaw -
+                  calculationData.startPositionValueRaw
+
+                if (
+                  this.getProtocolDetails().positionType === PositionType.Borrow
+                ) {
+                  profitRaw *= -1n
+                }
+
+                return {
+                  address,
+                  name,
+                  symbol,
+                  decimals,
+                  profitRaw,
+                  type: TokenType.Underlying,
+                  calculationData: {
+                    withdrawalsRaw: withdrawals[address] ?? 0n,
+                    withdrawals: formatUnits(
+                      withdrawals[address] ?? 0n,
+                      decimals,
+                    ),
+                    depositsRaw: deposits[address] ?? 0n,
+                    deposits: formatUnits(deposits[address] ?? 0n, decimals),
+                    startPositionValueRaw: startPositionValueRaw ?? 0n,
+                    startPositionValue: formatUnits(
+                      startPositionValueRaw ?? 0n,
+                      decimals,
+                    ),
+                    endPositionValueRaw,
+                    endPositionValue: formatUnits(
+                      endPositionValueRaw ?? 0n,
+                      decimals,
+                    ),
+                  },
+                }
+              },
+            ),
+          }
+        },
+      ),
+    )
+
+    return { tokens, fromBlock, toBlock }
+  }
+
+  async getMovements({
+    protocolToken,
+    underlyingTokens,
+    filter: { smartContractAddress, fromBlock, toBlock, from, to },
+  }: {
+    protocolToken: Erc20Metadata
+    underlyingTokens: Erc20Metadata[]
+    filter: {
+      smartContractAddress: string
+      fromBlock: number
+      toBlock: number
+      from?: string
+      to?: string
+    }
+  }): Promise<MovementsByBlock[]> {
+    const protocolTokenContract = MorphoAToken__factory.connect(
```

Comment:
```suggestion
    const protocolTokenContract = AToken__factory.connect(
```

Morpho is not tokeninzed so its missleading